booking.yaml

```diff
@@ -0,0 +1,516 @@
+      in: header
+  schemas:
+    error:
+      type: object
+      description: >-
+        An error that the service may send, e.g. in case of invalid input,
+        missing authorization or internal service error.
+      required:
+        - message
+        - code
+      properties:
+        message:
+          type: string
+          description: A human readable error message (preferrably in English)
+        code:
+          type: string
+          description: 'A TSP internal error code, used for reference'
+    newBooking:
+      type: object
+      description: 'A new booking, created by MaaS POST request in ''new'' state'
+      required:
+        - leg
+        - customer
+      properties:
+        state:
+          type: string
+          description: The state of the booking (always new for new bookings)
+          enum:
+            - NEW
+        leg:
+          $ref: '#/components/schemas/leg'
+        customer:
+          $ref: '#/components/schemas/customer'
+    booking:
+      type: object
+      description: >-
+        The booking information describing the state and details of the
+        transaction
+      allOf:
+        - $ref: '#/components/schemas/newBooking'
+      properties:
+        id:
+          description: The identifier MaaS will be using to referring to the booking
+          type: string
+        state:
+          $ref: '#/components/schemas/bookingState'
+        terms:
+          $ref: '#/components/schemas/bookingState'
+        token:
+          $ref: '#/components/schemas/token'
+        meta:
+          description: Arbitrary metadata that a TSP can add
+          type: object
+      required:
+        - id
+        - state
+        - leg
+        - customer
+        - token
+    bookingState:
+      description: The life-cycle state of the booking (from NEW to FINISHED)
+      type: string
+      enum:
+        - NEW
+        - BOOKED
+        - CANCELLED
+        - PAID
+        - UPDATE_REQUESTED
+        - UPDATED
+        - STARTED
+        - FINISHED
+    token:
+      description: >-
+        The validity token (such as booking ID, travel ticket etc.) that MaaS
+        clients will display to validate the trip when starting the leg.
+      properties:
+        validityDuration:
+          description: >-
+            The rules that MaaS will interpret to schedule, re-validate or
+            cancel the booking.
+          type: object
+          properties:
+            from:
+              description: The starting time from which the ticket is valid
+              $ref: '#/components/schemas/time'
+            to:
+              description: The finishing time the ticket is valid for
+              $ref: '#/components/schemas/time'
+        meta:
+          description: >-
+            Arbitrary metadata the TSP may pass along the ticket to the client
+            (e.g. a booking code, base64 encoded binary)
+          type: object
+    customer:
+      type: object
```

Comment:
This could refer to https://schema.org/Person and then use `givenName` and `familyName` (which is better for internationalisation and countries with different ordering of names)

Reply:
Added enhancement issue https://github.com/maas-alliance/apis/issues/14